mathesar/views/api.py

```diff
@@ -85,9 +89,19 @@ class RecordViewSet(viewsets.ViewSet):
 
     def list(self, request, table_pk=None):
         paginator = TableLimitOffsetPagination()
-        records = paginator.paginate_queryset(self.queryset, request, table_pk)
-        serializer = RecordSerializer(records, many=True)
-        return paginator.get_paginated_response(serializer.data)
+        filter_form = RecordListFilterForm(request.GET)
```

Comment:
Why use a form here?

Reply:
Forms and form fields are nice for the JSON input as they automatically parse and validate the url parameters for us. In this PR it's somewhat weird since there's only a single field, but its a bit more useful with the sorting and grouping fields added in. It is a little un-intuitive though - I'm happy to do it manually if this isn't a pattern we want to use.

Reply:
That's fine, add a comment explaining that in the code in case anyone else is confused. We can figure out a different pattern later if desired.